test/e2e/multicast_test.go

```diff
@@ -158,6 +170,553 @@ type multicastTestcase struct {
+		Spec: crdv1alpha1.ClusterNetworkPolicySpec{
+			AppliedTo: []crdv1alpha1.NetworkPolicyPeer{
+				{PodSelector: &selectorC},
+			},
+			Priority: p10,
+			Egress: []crdv1alpha1.Rule{
+				{
+					From: []crdv1alpha1.NetworkPolicyPeer{},
+					Protocols: []crdv1alpha1.NetworkPolicyProtocol{{
+						IGMP: &crdv1alpha1.IGMPProtocol{IGMPType: igmpType, GroupAddress: groupAddressDrop},
+					}},
+					Action: &dropAction,
+				},
+				{
+					From: []crdv1alpha1.NetworkPolicyPeer{},
+					Protocols: []crdv1alpha1.NetworkPolicyProtocol{{
+						IGMP: &crdv1alpha1.IGMPProtocol{IGMPType: igmpType, GroupAddress: groupAddressAllow},
+					}},
+					Action: &allowAction,
+				},
+			},
+		},
+	}
+	if _, err = k8sUtils.CreateOrUpdateACNP(acnp); err != nil {
+		failOnError(fmt.Errorf("create ACNP failed for ACNP %s: %v", acnp.Name, err), t)
+	}
+
+	// Wait for the policy to be realized before attempting connections
+	failOnError(data.waitForACNPRealized(t, acnp.Name), t)
+
+	go func() {
+		receiverMulticastDropCommand := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -o -p %d  -t 3 -c 1 -W %d %s", 2234, mcjoinWaitTimeout, groupAddressDrop)}
+		data.RunCommandFromPod(data.testNamespace, receiverName, mcjoinContainerName, receiverMulticastDropCommand)
+	}()
+	go func() {
+		receiverMulticastAllowCommand := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -o -p %d  -t 3 -c 1 -W %d %s", 2234, mcjoinWaitTimeout, groupAddressAllow)}
+		data.RunCommandFromPod(data.testNamespace, receiverName, mcjoinContainerName, receiverMulticastAllowCommand)
+	}()
+
+	if err := wait.Poll(5*time.Second, defaultTimeout, func() (bool, error) {
+		stdout, stderr, err := runAntctl(antreaPod, antctlCmds, data)
+		t.Logf("Got %s pod multicast stats output: %s", antreaPod, stdout)
+		if err != nil {
+			t.Errorf("Error when executing antctl get podmulticaststats, stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+			failOnError(err, t)
+		}
+		match, _ := regexp.MatchString(fmt.Sprintf("%s[[:space:]]+%s[[:space:]]+%d[[:space:]]+%d", data.testNamespace, receiverName, 0, 0), strings.TrimSpace(stdout))
+		if !match {
+			return false, nil
+		}
+		_, err = data.crdClient.StatsV1alpha1().MulticastGroups().Get(context.TODO(), groupAddressDrop, metav1.GetOptions{})
+		if err == nil {
+			return false, nil
+		}
+		multicastGroupAllow, err := data.crdClient.StatsV1alpha1().MulticastGroups().Get(context.TODO(), groupAddressAllow, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+		if len(multicastGroupAllow.Pods) != 1 || multicastGroupAllow.Pods[0].Name != receiverName {
+			return false, nil
+		}
+		stats, err := data.crdClient.StatsV1alpha1().AntreaClusterNetworkPolicyStats().Get(context.TODO(), "acnp1", metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		t.Logf("Got AntreaClusterNetworkPolicy stats: %v", stats)
+		if len(stats.RuleTrafficStats) != 2 {
```

Comment:
Maybe add a unique function to compare the number of stats.RuleTrafficStats, and check each rule has packets greater than 0?

Reply:
updated